### PR TITLE
TitleBar component

### PR DESCRIPTION
**Pull request by btzr-io**
- [x] Add View component.
-  ~Use toolbar instead of Navbar.~
- [x] Add TitleBar component.
- [ ] Toggle `max` button.
- [x] Update test.
- [x] Fix & tidy css.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 import Buttons  from './components/button';
 import Dropdowns  from './components/dropdown';
 import Switch  from './components/switch';
 import style from './style.styl';
 
+
 let Test = ({...props, t}) => (
     <div className={style.layout}>
+        {
+            props.TitleBar.map((i, k) =>
```
Comment (xaiki) [inline src/index.js]: this looks wrong, props should be lowercase

File: src/index.js [hunk 2]
```diff
@@ -22,10 +31,15 @@ let Test = ({...props, t}) => (
         </div>
 
         <div className={style.test}>
-            {props.Dropdowns.map((i, k) => <Dropdowns.Dropdown key={k} {...i}/>)}
+            {props.Dropdowns.text.map((i, k) => <Dropdowns.Dropdown key={k} {...i}/>)}
```
Comment (xaiki) [inline src/index.js]: props should be lowercase

File: src/index.js [hunk 3]
```diff
@@ -22,10 +31,15 @@ let Test = ({...props, t}) => (
...
         </div>
 
         <div className={style.test}>
+            {props.Dropdowns.color.map((i, k) => <Dropdowns.DropdownColor key={k} {...i}/>)}
```
Comment (xaiki) [inline src/index.js]: ditto

File: test/data.js [hunk 4]
```diff
@@ -21,8 +51,8 @@ export default {
             action: () => {}
         },
     ],
-    Dropdowns: [
-        {
+    Dropdowns: {
+        text: [{
```
Comment (xaiki) [inline test/data.js]: why not make a list of dropdowns object with their type encoded ?

Comment (xaiki) [reply test/data.js]: what does this have to do with this patch ?

**Review by xaiki — CHANGES_REQUESTED** [review]
a few nitpicks

**Comment by xaiki** [comment]
merging navbar and toolbar seems odd, why would we want to re-render the whole navbar when we change a state in the toolbar ?

**Comment by btzr-io** [comment]
Navbar was just a wrapper for toolbar...
Why would we want to re-render it? ^^
`toolbar = navbar + action buttons`

File: test/data.js (hunk 4 above)
Comment (btzr-io) [reply test/data.js]: that's for `test` ( preview components )

**Comment by btzr-io** [comment]
Anyway, we would need to use redux or flux to make everything work

File: src/index.js (hunk 1 above)
Comment (btzr-io) [reply src/index.js]: ok, fixed

**Comment by xaiki** [comment]
> Navbar was just a wrapper for toolbar...
> Why would we want to re-render it? ^^
> `toolbar = navbar + action buttons`

because components re-render when props change
because pure-components check for their prop-changes before re-rendering.

**Comment by btzr-io** [comment]
Ok, we also need `toolbar` outside `navbar` :upside_down_face: ^^

**Comment by xaiki** [comment]
looks good to me, do you need to fix the max button ?

**Comment by btzr-io** [comment]
yes but I don't know how to do it yet :stuck_out_tongue:

**Comment by btzr-io** [comment]
### Nwjs
``` JS
win.on('maximize', function() {
  // make react component listen...
  trigger.event()
});
win.on('unmaximize', function() {
  // make react component listen...
  trigger.event()
});
```
### React
``` JS
onMaximize () {
    this.setState({ max: true })
}
```

**Comment by btzr-io** [comment]
but we aren't gonna use `title-bar` yet , so you can merge it
@xaiki ^^

File: index.js [hunk 5]
```diff
@@ -10,4 +11,4 @@ import Modal        from './src/components/modal'
 // XXX: backward compat
 let ActionBar = Navbar
```
Comment (btzr-io) [inline index.js]: what's that for? ^^

Comment (xaiki) [reply index.js]: backward compatibility,
you should probably remove that now.
we use to call ActionBar Navbar, and the rest of the code depended on it.
now that you have a real navbar you should probably remove that hack.

File: index.js [hunk 6]
```diff
@@ -1,13 +1,11 @@
 import React, { Component } from 'react';
```
Comment (btzr-io) [inline index.js]: We don't need to import react ^^

Comment (xaiki) [reply index.js]: are you sure ? did you try it without it ? i think the import react tweaks some bits in the import/export process.

Comment (btzr-io) [reply index.js]: yes, I don't see any errors :confused:

Comment (btzr-io) [reply index.js]: > the import react tweaks some bits in the import/export process.

so do we need it or not? ^^

Comment (xaiki) [reply index.js]: if you tested it and it works, we don't =P

**Pull request merged**